### PR TITLE
various: fixes for strictDeps = true

### PR DESCRIPTION
**Pull request by FliegendeWurst**
ref. #178468

I have also increased the test timeout for the dbus python package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/dbus-python/default.nix [hunk 1]
```diff
@@ -90,7 +90,7 @@ lib.fix (
     checkPhase = ''
       runHook preCheck
 
-      meson test -C _meson-build --no-rebuild --print-errorlogs
+      meson test -C _meson-build --no-rebuild --print-errorlogs --timeout-multiplier 0
```
Comment (paparodeo) [inline pkgs/development/python-modules/dbus-python/default.nix:93]:
```suggestion
      meson test -C _meson-build --no-rebuild --print-errorlogs
```

not required due to https://github.com/NixOS/nixpkgs/pull/356589

**Review by paparodeo — COMMENTED** [review]
why not also add `strictDeps = true;` ?

File: pkgs/development/python-modules/dbus-python/default.nix (hunk 1 above)
Comment (FliegendeWurst) [reply pkgs/development/python-modules/dbus-python/default.nix:93]: It is required. That PR only modifies the default mesonFlags, this is a custom meson invocation.

**Comment by FliegendeWurst** [comment]
> why not also add `strictDeps = true;` ?

I have seen some silent breakage in e.g. pulseaudio when setting that, so I was a bit hesistant to just add it. But I suppose we'll see if it breaks anything for these packages.

File: pkgs/by-name/li/libmysqlconnectorcpp/package.nix
Comment (philiptaron) [inline pkgs/by-name/li/libmysqlconnectorcpp/package.nix:1]: Cherry-picked 65edc93a868149e65d16c10369b882a98eaa1766 against master and built `libmysqlconnectorcpp`. ✅

**Review by philiptaron — APPROVED** [review]
Thank you for your contribution, @FliegendeWurst. 

This PR would have been substantially easier to review if it were one PR per package, and would have enabled many of this PRs to target master instead of staging. I would appreciate that in the future changes in service of `strictDeps = true`.

**Pull request merged**